src/Dockerfile

```
@@ -83,11 +83,12 @@ RUN if   [ "$TARGETPLATFORM" = "linux/amd64" ];    then FTLARCH=amd64; \
     && echo "Arch: ${TARGETPLATFORM}, FTLARCH: ${FTLARCH}" \
     && curl -sSL "https://ftl.pi-hole.net/${FTL_BRANCH}/pihole-FTL-${FTLARCH}" -o /usr/bin/pihole-FTL \
     && chmod +x /usr/bin/pihole-FTL \
-    && readelf -h /usr/bin/pihole-FTL || cat /usr/bin/pihole-FTL
+    && readelf -h /usr/bin/pihole-FTL || (cat /usr/bin/pihole-FTL && echo "Error with downloaded FTL binary" && exit 1)
```

Do we need the full file contents?

Maybe using `head -n 5` is enough and avoid too many lines if the downloaded file has many lines:
```suggestion
    && readelf -h /usr/bin/pihole-FTL || (head -n 5 /usr/bin/pihole-FTL && echo "Error with downloaded FTL binary" && exit 1)
```

---

Seems a decent compromise - @DL6ER , any thoughts?

---

I'm still feeling we should not try to print at all, we just don't what is there

---

Agree... It could be a binary file with long lines.